src/extensions/replay/sessionrecording.ts

```diff
@@ -459,28 +580,52 @@ export class SessionRecording {
         return url
     }
 
+    private clearBuffer(): SnapshotBuffer {
+        this.buffer = undefined
+
+        return {
+            size: 0,
+            data: [],
+            sessionId: this.sessionId,
+            windowId: this.windowId,
+        }
+    }
+
+    // the intention is a buffer that (currently) is used only after a decide response enables session recording
+    // it is called ever X seconds using the flushBufferTimer so that we don't have to wait for the buffer to fill up
+    // when it is called on a timer it assumes that it can definitely flush
+    // it is flushed when the session id changes or the size of the buffered data gets too great (1mb by default)
+    // first change: if the recording is in buffering mode,
+    //  flush buffer simply resets the timer and returns the existing flush buffer
     private _flushBuffer() {
         if (this.flushBufferTimer) {
             clearTimeout(this.flushBufferTimer)
             this.flushBufferTimer = undefined
         }
 
+        const minimumDuration = this.minimumDuration
+        const bufferedDuration = this.bufferedDuration
+        const isBelowMinimumDuration =
+            _isNumber(minimumDuration) && _isNumber(bufferedDuration) && bufferedDuration < minimumDuration
+
+        if (this.status === 'buffering' || isBelowMinimumDuration) {
+            this.flushBufferTimer = setTimeout(() => {
+                this._flushBuffer()
+            }, RECORDING_BUFFER_TIMEOUT)
+            return this.buffer || this.clearBuffer()
+        }
```

Comment:
This addition comes with the new challenge of the buffer getting too big... If we buffer 15 seconds and then flush, it will essentially always buffer one super big event. Which is probably fine for now but would be interesting to see if that becomes an issue...

Reply:
Yeah, I backed away from the buffer until manual flush change in this PR. Partly cos it is getting massive but also because I think the flushing can be improved now we know we have checkout snapshots on an interval. We should be able to walk the buffer between those checkouts sending "complete" sections.

But the more we change here the harder it is to understand what might break 🙈